pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/SchemaDefinition.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api.schema;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A schema definition
+ * {@link org.apache.pulsar.client.api.Schema} for the schema definition value.
+ */
+@Data
+@EqualsAndHashCode
+@ToString
+public class SchemaDefinition<T> {
+
+    /**
+     * the schema definition class
+     */
+    private final Class<T> clazz;
+    /**
+     * The flag of schema type always allow null
+     */
+    private boolean alwaysAllowNull = true;
+    /**
+     * The schema info properties
+     */
+    private Map<String, String> properties = new HashMap<>();
+
+
+    public SchemaDefinition(Class<T> clazz) {
+
+        properties.put("__alwaysAllowNull", "true");
+        this.clazz = clazz;
+
+    }
+
+    /**
+     * Set schema whether always allow null or not
```

Comment:
We should explain here what are the implications of using allowNull=true/false in the Avro context and indicate what is the default behavior.